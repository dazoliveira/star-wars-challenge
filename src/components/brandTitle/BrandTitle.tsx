import { Grid, Typography } from "@material-ui/core";

type BrandTitleProps = {
    welcomeText: string
    subTitle: string
    brandName: string
}

export default function BrandTitle({ welcomeText, subTitle, brandName }: BrandTitleProps) {

    return (    
        <Grid 
          direction="column"
          justifyContent="center"
          alignItems="center"
        >
          <Grid item xs={12}>
            <Typography variant='h2' color='primary' align='center' gutterBottom>
              {welcomeText} <Typography variant='h1' component='span'>{brandName}</Typography>
            </Typography>
          </Grid>
          <Grid item xs={12}>
            <Typography color='primary' variant='subtitle2' align='center'>
                {subTitle}
            </Typography>
          </Grid>

        </Grid>
    )
}