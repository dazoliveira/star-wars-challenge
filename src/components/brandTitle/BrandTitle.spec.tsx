import { screen } from '@testing-library/react'
import { renderWithTheme } from "utils/tests/helpers";
import BrandTitle, { BrandTitleProps } from './BrandTitle'

let props:BrandTitleProps = {
    welcomeText: 'Welcome to' , 
    brandName: 'IClinic',
    subTitle: 'frontend test', 
}

describe('<BrandTitle />', () => {

    it('should render brand title', () => {

        const { container } = renderWithTheme(<BrandTitle {...props }/>)

        expect(screen.queryByTestId('brand-title')).toBeInTheDocument()

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render welcome text', () => {

        renderWithTheme(<BrandTitle {...props }/>)

        expect(screen.queryByText('Welcome to')).toBeInTheDocument()
    })

    it('should render brand name text', () => {

        renderWithTheme(<BrandTitle {...props }/>)

        expect(screen.queryByText('IClinic')).toBeInTheDocument()
    })

    it('should render brand subtitle text', () => {

        renderWithTheme(<BrandTitle {...props }/>)

        expect(screen.queryByText('frontend test')).toBeInTheDocument()
    })
});