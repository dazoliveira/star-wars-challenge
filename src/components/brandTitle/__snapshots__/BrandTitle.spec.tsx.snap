// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrandTitle /> should render brand title 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
  data-testid="brand-title"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 MuiTypography-colorPrimary MuiTypography-gutterBottom MuiTypography-alignCenter"
    >
      Welcome to
       
      <span
        class="MuiTypography-root MuiTypography-h1"
      >
        IClinic
      </span>
    </h2>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <p
      class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorPrimary MuiTypography-alignCenter"
    >
      frontend test
    </p>
  </div>
</div>
`;
